xmake/modules/package/manager/cmake/find_package.lua

```diff
@@ -81,25 +81,39 @@ function _find_package(cmake, name, opt)
     cmakefile:print("find_package(%s REQUIRED %s)", requirestr, componentstr)
     cmakefile:print("if(%s_FOUND)", name)
     cmakefile:print("   add_executable(%s test.cpp)", testname)
-    cmakefile:print("   target_include_directories(%s PRIVATE ${%s_INCLUDE_DIR} ${%s_INCLUDE_DIRS})",
-        testname, name, name)
-    cmakefile:print("   target_include_directories(%s PRIVATE ${%s_INCLUDE_DIR} ${%s_INCLUDE_DIRS})",
-        testname, name:upper(), name:upper())
+    -- setup include directories
+    local includedirs = ""
+    if configs.include_directories then
+        includedirs = table.concat(table.wrap(configs.include_directories), " ")
+    else
+        includedirs = ("${%s_INCLUDE_DIR} ${%s_INCLUDE_DIRS}"):format(name, name)
+        includedirs = includedirs .. (" ${%s_INCLUDE_DIR} ${%s_INCLUDE_DIRS}"):format(name:upper(), name:upper())
+    end
+    cmakefile:print("   target_include_directories(%s PRIVATE %s)", testname, includedirs)
+    -- reserved for backword compatibility
     cmakefile:print("   target_include_directories(%s PRIVATE ${%s_CXX_INCLUDE_DIRS})",
         testname, name)
-    cmakefile:print("   target_link_libraries(%s ${%s_LIBRARY} ${%s_LIBRARIES} ${%s_LIBS})",
-        testname, name, name, name)
-    cmakefile:print("   target_link_libraries(%s ${%s_LIBRARY} ${%s_LIBRARIES} ${%s_LIBS})",
-        testname, name:upper(), name:upper(), name:upper())
+    -- setup link library/target
+    local linklibs = ""
     if configs.link_libraries then
-        cmakefile:print("   target_link_libraries(%s %s)",
-            testname, table.concat(table.wrap(configs.link_libraries), " "))
+        linklibs = table.concat(table.wrap(configs.link_libraries), " ")
+    else
+        linklibs = ("${%s_LIBRARY} ${%s_LIBRARIES} ${%s_LIBS}"):format(name, name, name)
+        linklibs = linklibs .. (" ${%s_LIBRARY} ${%s_LIBRARIES} ${%s_LIBS}"):format(name:upper(), name:upper(), name:upper())
     end
+    cmakefile:print("   target_link_libraries(%s PRIVATE %s)", testname, linklibs)
     cmakefile:print("endif(%s_FOUND)", name)
     cmakefile:close()
 
     -- run cmake
     local envs = configs.envs or opt.envs
+    if (configs.debug) then
+        if (not envs) then
+            envs = {CMAKE_BUILD_TYPE = "Debug"}
+        elseif (not envs.CMAKE_BUILD_TYPE) then
+            envs.CMAKE_BUILD_TYPE = "Debug"
+        end
```

Comment:
just only two lines

```
envs = envs or {}
envs.CMAKE_BUILD_TYPE = envs.CMAKE_BUILD_TYPE or "Debug"
```

Reply:
It confuses me, earlier you said set `envs` to `{}` would overwrite the whole envs (see [this](https://github.com/xmake-io/xmake/pull/3493/files/001e4781aaaea799f53a4ab76903ba3930ad1b24#diff-da48836594df046b6b3465cf6d19e24de62a47e2b978f210d246842706506922R109)). Only a `nil` or a table with value is accepted?

Reply:
envs has always values if it's debug mode, so we can set `envs = envs or {}`, because it is always non-empty.

but if it's release mode, envs cannot be `{}`